components/Animation.js

```diff
@@ -0,0 +1,45 @@
+const React = require('react');
+
+class Animation extends React.Component {
+
+  // TODO: create a componentWillUpdate() here
+  componentWillUpdate() {
+    this.showLoadingBar();
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      url: ' http://placehold.it/500x150'
+    };
+    this.showLoadingBar = this.showLoadingBar.bind(this);
+  }
+
+  getNewCat() {
+    fetch('http://api.giphy.com/v1/gifs/random?api_key=dc6zaTOxFJmzC')
+      .then((res, err) => { 
+        if (err) {
+          console.log('Something went wrong with fetching your new cat!', err)
+        } else {
+          res.json().then(result => this.setState({ url: result.data.fixed_height_downsampled_url }));
+        }
+      });
+  }
+
+  showLoadingBar() {
+    const progressBar = document.getElementById('progress-bar');
+    progressBar.className = 'off on';
+    setTimeout(() => progressBar.className = 'off', 3000);
+  }
+
+  render() {
+    return (
+      <div>
+        <img src={this.state.url} height="100px"/>
+        <div><button onClick={() => this.getNewCat()}>New random .gif!</button></div>
```

Comment:
These are equivalent - using an arrow function will work as a bind so you don't have to do it in the constructor.

```
<button onClick={() => this.getNewCat()}
```
is equivalent to
```
<button onClick={this.getNewCat().bind(this)}
```
which is equivalent to
```
getNewCat = () => {
  {---}
}
<button onClick={this.getNewCat()}
```
which is equivalent to
```
constructor{
  this.getNewCat = this.getNewCat.bind(this);
}

<button onClick={this.getNewCat()}
```

I prefer to use arrow functions when possible, but I'll change it to what you suggested if it's what's preferred in the project! :)